wgpu-hal/src/dynamic/mod.rs

```diff
@@ -0,0 +1,67 @@
+use std::any::Any;
+
+use wgt::WasmNotSendSync;
+
+/// Base trait for all resources, allows downcasting via `std::any::Any`.
+pub trait DynResource: Any + WasmNotSendSync + 'static {
```

Comment:
I'm also not a fan of the `Dyn` prefix. What is the rationale behind it?

Reply:
Felt like I had to prefix it with _something_ because just distinguishing by namespace gets confusing quickly imho and that's what I came up with 🤷 
open for suggestions!

Reply:
I haven't looked at the PRs as others are reviewing them but don't these traits represent the full surface of the hal API? If so, I don't think we need a prefix at all.

Reply:
It's almost the entire surface, yes. I think once the refactor is done they are no longer needed. The thing that I found them useful was probably more transitional!

Reply:
Ah there's some name clashes though between the generic traits that HAL has. E.g. there's `Device<A>` and `DynDevice` 🤔

Reply:
Oh so we still have a `Device<A>`? I should just take a look at the code.

Reply:
oops. just deleted my previous comment because I thought wait this is silly and surely nobody read it yet:
Yes we still have the `Device<A>` baseclass, all this stuff hasn't changed. All individual hal implementors still do the same thing. But that nameclash isn't reaaaaallly a reason not to have this. I mean dx12 also has its own `Device`

Reply:
We do however force everything on `Api` now to implement (e.g) `DynDevice` and then say that all `Device<A>` blanket implement `DynDevice`

Reply:
I see, well we can remove the prefix in a follow-up.